assets/src/components/customers/CustomerDetailsSidebar.tsx

```diff
@@ -0,0 +1,316 @@
+            Icon={DesktopOutlined}
+            name="Browser"
+            value={browser}
+          />
+          <CustomerDetailsProperty
+            Icon={DesktopOutlined}
+            name="OS"
+            value={os}
+          />
+          <CustomerDetailsProperty
+            Icon={DesktopOutlined}
+            name="IP"
+            value={lastIpAddress}
+          />
+        </CustomerDetailsSection>
+
+        <Divider dashed />
+
+        <CompanyDetailsSection company={company} />
+
+        <Divider dashed />
+
+        {hasMetadata && (
+          <>
+            <CustomerDetailsSection title="Metadata">
+              {Object.entries(metadata).map(([key, value]) => (
+                <CustomerDetailsProperty
+                  Icon={InfoCircleOutlined}
+                  key={key}
+                  name={key}
+                  value={value}
+                />
+              ))}
+            </CustomerDetailsSection>
+            <Divider dashed />
+          </>
+        )}
+
+        <CustomerDetailsSection title="Tags">
+          <SidebarCustomerTags customerId={customerId} />
+        </CustomerDetailsSection>
+      </Box>
+    </CustomerDetailsCard>
+  );
+};
+
+const CompanyDetailsSection = ({company}: {company: Company | undefined}) => {
+  let body;
+
+  if (company) {
+    const {
+      id: companyId,
+      name,
+      website_url: websiteUrl,
+      slack_channel_id: slackChannelId,
+      slack_channel_name: slackChannelName,
+    } = company;
+
+    body = (
+      <>
+        <CustomerDetailsProperty
+          Icon={TeamOutlined}
+          name="Name"
+          value={<Link to={`/companies/${companyId}`}>{name}</Link>}
+        />
+        <CustomerDetailsProperty
+          Icon={LinkOutlined}
+          name="Website"
+          value={websiteUrl}
+        />
+        <CustomerDetailsProperty
+          Icon={LinkOutlined}
+          name="Slack Channel"
+          value={
+            slackChannelId &&
+            slackChannelName && (
+              <a
+                href={`https://slack.com/app_redirect?channel=${slackChannelId}`}
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                {slackChannelName}
+              </a>
+            )
+          }
+        />
+      </>
+    );
+  } else {
+    // TODO: add ability to link customer to company
+    body = <Text>Customer is not linked to a company.</Text>;
+  }
+
+  return (
+    <CustomerDetailsSection title="Company">{body}</CustomerDetailsSection>
+  );
+};
+
+export const CustomerDetailsSection = ({
+  title,
+  children,
+}: {
+  title: string;
+  children: React.ReactNode;
+}) => {
+  return (
+    <Box>
+      <Box mb={2}>
+        <Text strong>{title}</Text>
+      </Box>
+      {children}
+    </Box>
+  );
+};
+
+const CustomerDetailsProperty = ({
+  Icon,
+  name,
+  value,
+}: {
+  Icon: React.ComponentType<any>;
+  name: string;
+  value: any;
+}) => {
+  let valueComponent;
```

Comment:
hmm i wonder if similar to the `icon` prop we should just enforce that this is a `ReactElement` type and explicitly pass in `<Text>{value || 'Unknown'}</Text>` when we need to 🤔 i kinda prefer the explicit nature of it, and in some cases we might want the fallback value to be something other than "unknown" 

(this behavior might also be confusing if a value of `0` is passed in, for example... though i suppose this edge case could be fixed by using `value ?? 'Unknown'` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)

otherwise, small nitpick but we can probably just assign the `valueComponent` in a ternary expression, e.g.
```js
const valueComponent = React.isValidElement(value) ? value : <Text>{value || 'Unknown'}</Text>;
```

Reply:
I considered that as well. I had it set up this way to reduce code duplication, but I'm not actually a huge fan of code paths where different types render different results (harder to reason about). Let's err on the side of being explicit. I'll change it 👌